plugin-api/src/main/java/org/hyperledger/besu/plugin/services/PermissioningService.java

```diff
@@ -14,6 +14,7 @@
  */
 package org.hyperledger.besu.plugin.services;
 
+import org.hyperledger.besu.ethereum.permissioning.account.TransactionPermissioningProvider;
```

Comment:
will need some refactoring to move this TransactionPermissioningProvider interface into the plugin package.

Reply:
Thank you for your feedback! Are you referring to this location: `plugin-api/src/main/java/org/hyperledger/besu/plugin/services/permissioning/`?

Reply:
I have a quick question: Since the TransactionPermissioningProvider is implemented by both the AccountLocalConfigPermissioningController and the TransactionSmartContractPermissioningController, what would be the best way to move forward with this? Your guidance would be greatly appreciated!

Reply:
I would have a look at how `NodeMessagePermissioningProvider` (functional interface) is used, since that's already exposed to plugins and used elsewhere in besu. It's a common pattern in besu to have an interface in the plugin-api package which is imported and implemented by classes where needed. `BlockHeader` is another good example but you can find lots of examples in the plugin-api package.

Reply:
There seems to be a circular dependency issue caused by importing the Transaction class into the TransactionPermissioningProvider interface.  introducing a lightweight abstraction (like a DTO or a subset of transaction properties) for permissioning logic might be a good solution. wdyt?

Reply:
does it help if the TransactionPermissioningProvider uses the org.hyperledger.besu.datatypes.Transaction interface instead

Reply:
Thank you very much. It works perfectly!.